docs/gitalb/gitlab-gcp.md

```diff
@@ -0,0 +1,380 @@
+---
+Generate PostgreSQL secret and values file
+secret/gitlab-postgresql created
+---
+Generate Redis values file
+---
+Generate Buckets secrets
+secret/bucket-artifacts created
+secret/bucket-backups-tmp created
+secret/bucket-backups created
+secret/bucket-externaldiffs created
+secret/bucket-lfs created
+secret/bucket-packages created
+secret/bucket-pseudonymizer created
+secret/bucket-registry created
+secret/bucket-uploads created
+
+``` 
+
+## Install
+Render the official GitLab Helm chart with  generated values files, and your settings into a `gitlab-gcp.yaml` file.
+See [GitLab Helm Documentation](https://docs.gitlab.com/charts/installation/deployment.html) for the additional details
+
+```bash
+helm repo add gitlab https://charts.gitlab.io/
+helm repo update
+helm fetch gitlab/gitlab
+helm template gitlab-1.8.4.tgz --name gitlab-demo --namespace gitlab \
+    -f cluster/examples/gitlab/gcp/values-buckets.yaml \
+    -f cluster/examples/gitlab/gcp/values-redis.yaml \
+    -f cluster/examples/gitlab/gcp/values-psql.yaml \
+    --set global.hosts.domain=your.domain \
+    --set global.hosts.hostSuffix=demo \
+    --set certmanager-issuer.email=email@account.io > gitlab-gcp.yaml
+```
+
+Examine `gitlab-gcp.yaml` to familiarize yourself with all GitLab components.
+
+Install GitLab
+Note: your output may look differently:
+```bash
+kubectl create -f gitlab-gcp.yaml
+```
+
+## Uninstall
+
+### GitLab
+To remove the GitLab application from the GKE cluster: run:
+```bash
+kubectl delete -f gitlab-gcp.yaml
+```
+
+### Crossplane
+To remove Crossplane managed resources, switch back to local cluster context `minikube`:
+```bash
+kubectl config use-context minikube
+```
+
+Delete all managed resources by running:
+```bash
+kubectl delete -Rf cluster/examples/gitlab/gcp/resource-claims
+---
+bucket.storage.crossplane.io "gitlab-artifacts" deleted
+bucket.storage.crossplane.io "gitlab-backups-tmp" deleted
+bucket.storage.crossplane.io "gitlab-backups" deleted
+bucket.storage.crossplane.io "gitlab-externaldiffs" deleted
+bucket.storage.crossplane.io "gitlab-lfs" deleted
+bucket.storage.crossplane.io "gitlab-packages" deleted
+bucket.storage.crossplane.io "gitlab-pseudonymizer" deleted
+bucket.storage.crossplane.io "gitlab-registry" deleted
+bucket.storage.crossplane.io "gitlab-uploads" deleted
+kubernetescluster.compute.crossplane.io "gitlab-gke" deleted
+postgresqlinstance.storage.crossplane.io "gitlab-postgresql" deleted
+rediscluster.cache.crossplane.io "gitlab-redis" deleted
+```
+
+Delete all resource classes:
+```bash
+kubectl delete -Rf cluster/examples/gitlab/gcp/resource-classes/
+---
+resourceclass.core.crossplane.io "standard-gcp-bucket" deleted
+resourceclass.core.crossplane.io "standard-gcp-cluster" deleted
+resourceclass.core.crossplane.io "standard-gcp-postgres" deleted
+resourceclass.core.crossplane.io "standard-gcp-redis" deleted
+```
+
+Delete gcp-provider and secrets
```

Comment:
should we have a step that tells them to wait for all the external resources to get cleaned up before deleting the provider objects?